### PR TITLE
Remove _copy_construct factory

### PR DESCRIPTION
**Pull request by vyasr**
This PR removes the `_copy_construct` factory method for the `Frame` types that were still exposing it, replacing its usage with `_from_data`. The current implementation of `_from_data` is slightly faster, and more importantly this change leaves us with a single fast path for building `Frame` objects (bypassing the slow constructors) so that we only have to maintain and optimize one going forward.

### CODE REVIEW

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -3009,9 +3008,11 @@ def astype(self, dtype, copy=False, errors="raise"):
             return self.copy(deep=copy)
         try:
             data = self._column.astype(dtype)
+            if copy:
+                data = data.copy(deep=True)
 
-            return self._copy_construct(
-                data=data.copy(deep=True) if copy else data, index=self.index
+            return self._from_data(
+                {self.name: data.copy(deep=True)}, index=self._index
```
Comment (marlenezw) [inline python/cudf/cudf/core/series.py:3015]: Is this still ok to do if `copy` is False?

**Comment by marlenezw** [comment]
Looks really great to me overall! These changes are really good 😄

File: python/cudf/cudf/core/series.py (hunk 1 above)
Comment (vyasr) [reply python/cudf/cudf/core/series.py:3015]: Great catch! I had moved the copy logic into a separate line there and forgotten to re-inline it properly, thanks.

**Comment by vyasr** [comment]
rerun tests

**Review by marlenezw — APPROVED** [review]
Looks good to me!

**Comment by vyasr** [comment]
@gpucibot merge

**Pull request merged**